saleor/graphql/core/mutations.py

```diff
@@ -316,24 +342,52 @@ def check_permissions(cls, context, permissions=None):
 
         The `context` parameter is the Context instance associated with the request.
         """
-        permissions = permissions or cls._meta.permissions
-        if not permissions:
-            return True
-        if context.user.has_perms(permissions):
+        all_permissions = permissions or cls._meta.permissions
+        if not all_permissions:
             return True
+
+        permission_fns = [
+            p for p in all_permissions if isinstance(p, PermissionFunctions)
+        ]
+        admin_permissions = [
```

Comment:
Do these names are enough meaningful? 
- `permission_fns` - it is a list of permission that determines access to resources by a different states, right? Like `user is authorized`, `app is authorized`, `customer is authorized`
- `admin_permissions` -  Does it contain a list of our permission enums, like `manage_orders`? Does `admin_` gives us any context here? Personally I thought that `admin_permissions` will be used only for users that are staff but I see that we also check if `app` has `admin_permissions`. Not sure if using `admin_` is ok here, as `App` cannot be an admin, right?


Maybe we could give more context for both of these values?

Reply:
You guessed right about both names and their purpose. I didn't have any better idea how to generally call the fact of using "manage_" and "handle_" enums for permission checks and I called them "admin", but you're right this may suggest a physical user while it also applies to apps. I'm happy to accept better ideas for names here.

Reply:
`permission_fns` -> maybe something like `access_permissions`? As it contains all possible access permissions for the requestor.
`admin_permissions` -> `system_permissions` or something that will tell us that it contains a list of our built-in permissions.

Reply:
Changed:
permission_fns -> authorization_filters
admin_permissions -> permissions